src/components/Api/Api.jsx

```diff
@@ -0,0 +1,295 @@
+import React from 'react';
+import fetch from 'isomorphic-fetch';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Helmet from 'react-helmet';
+import CircularProgress from 'material-ui/CircularProgress';
+import RaisedButton from 'material-ui/RaisedButton';
+import styled from 'styled-components';
+import StripeCheckout from 'react-stripe-checkout';
+import moment from 'moment';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn,
+} from 'material-ui/Table';
+import strings from '../../lang';
+
+const URI = '/api/keys';
+
+const ApiContainer = styled.div`
+  width: 80%;
+  margin: 0 auto;
+  
+  & li {
+    list-style-type: initial;
+  }
+`;
+
+const KeyContainer = styled.pre`
+  background: grey;
+  display: inline;
+  padding: 10px;
+`;
+
+class KeyManagement extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      error: false,
+      loading: true,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`${process.env.REACT_APP_API_HOST}${URI}`, {
+      credentials: 'include',
+      method: 'GET',
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else if (res.status === 403) {
+          return {};
+        }
+        throw Error();
+      })
+      .then((json) => {
+        const update = json;
+        update.loading = false;
+        this.setState(update);
+      })
+      .catch(() => this.setState({ error: true }));
+  }
+
+  handleSubmit(token) {
+    this.setState({ loading: true });
+    fetch(`${process.env.REACT_APP_API_HOST}${URI}`, {
+      credentials: 'include',
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        token,
+      }),
+    })
+      .then((res) => {
+        if (res.ok) {
```

Comment:
I don't think you need the error logic here, if it's a bad request it should hit the catch block automatically

Reply:
I tried this and it doesn't work. See  https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch (See Checking that the fetch was successful)

Reply:
In that case our retry logic across the board probably should be fixed, in `action.js`